src/main/java/flaxbeard/immersivepetroleum/common/blocks/metal/TileEntityPumpjack.java

```diff
@@ -172,7 +198,7 @@ public void update(boolean consumePower)
 		int consumed = IPConfig.Extraction.pumpjack_consumption;
 		int extracted = consumePower ? energyStorage.extractEnergy(consumed, true) : consumed;
 				
-		if (extracted >= consumed && canExtract() && !this.isRSDisabled())
+		if (extracted >= consumed && canExtract() && !this.isRSDisabled() && hasPipe())
```

Comment:
You are running your code every tick. This is not particularly performant. I'd recommend checking only every few ticks (i.e. every 5 seconds, 100 ticks) if the pipe is still there. Besides that, why not move the hasPipe() function into the canExtract() function?